### PR TITLE
fix: handle notification api calls settings

### LINKED ISSUE #28173: [Bug]: Notifications - When Onboarding for the first time I get a the errors: `User storage does not exist` `Failed to check accounts presence`
### Describe the bug

Whenever I am Onboarding to the wallet for the first time (both Importing an SRP or Creating a Wallet), when I land into the Home page and go to the Notifications Settings, I see a couple of networks request fail and get the console errors:
- `User storage does not exist`
- `Failed to check accounts presence`

The failed network errors are:
- GET https://user-storage.api.cx.metamask.io/api/v1/userstorage/notifications/KEY --> 404 not found {"error":"key not found","message":"cannot get key"}

I was using the prod [dist build from circle ci](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/108487/workflows/cb4bd2ef-d019-47fb-ad4f-ee89aabc26ce/jobs/4057321/artifacts), so all the envars should be correct

### Expected behavior

_No response_

### Screenshots/Recordings

https://github.com/user-attachments/assets/1fb02b0a-83a2-495f-8c9d-1f2ca562c20f

### Steps to reproduce

1. Onboard for the first time
2. Land into Home page
3. Click Settings / Notifications
4. See errors and failed network requests

### Error messages or log output

_No response_

### Detection stage

On the development branch

### Version

12.8.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Adds effect checks; removes old hooks unused; and adds tests.

I really HATE the useFootguns we have everywhere for notifications. I want us to do a wider cleanup task.

1. Remove the Provider values. We can keep the provider to run initial effects, but other than that, we should not expose anything from it!
2. Remove nearly all of the useEffects scattered across all the components. Instead we can utilise 1 reusable hook for fetching.
   - potentially we can expose this in the provider if we really want to fetch only once!

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29327?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/28173

## **Manual testing steps**

1. Go through onboarding, and settings pages > Notifications
2. Check console and network tab, are we making API calls and throwing errors in console?

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by mathieuartu — APPROVED** [review]
LGTM!

File: ui/hooks/metamask-notifications/useSwitchNotifications.test.tsx [hunk 1]
```diff
@@ -1,125 +1,183 @@
...
-  deleteOnChainTriggersByAccount: jest.fn(),
-  showLoadingIndication: jest.fn(),
-  hideLoadingIndication: jest.fn(),
-  fetchAndUpdateMetamaskNotifications: jest.fn(),
-}));
-
-describe('useSwitchNotifications', () => {
-  let store: Store;
-
+describe('useSwitchFeatureAnnouncementsChange() tests', () => {
```
Comment (Prithpal-Sooriya) [inline ui/hooks/metamask-notifications/useSwitchNotifications.test.tsx:11]: Tests were a bit messy and didn't cover all cases.

File: ui/hooks/metamask-notifications/useSwitchNotifications.test.tsx [hunk 2]
```diff
@@ -1,125 +1,183 @@
...
+      mockSetFeatureAnnouncementsEnabled,
+    };
+  };
+
+  it('should update feature announcement when callback invoked', async () => {
+    const mocks = arrangeMocks();
+    const hook = renderHookWithProviderTyped(
+      () => useSwitchFeatureAnnouncementsChange(),
+      {},
+    );
```
Comment (Prithpal-Sooriya) [inline ui/hooks/metamask-notifications/useSwitchNotifications.test.tsx:31]: replaced mocking store and dispatch to instead use `renderHookWithProviderTyped` which does the setup for you.

The `{}` in the second param is to let the function add redux provider (we could instead pass in a partial store, but it is unneeded)

File: ui/hooks/metamask-notifications/useSwitchNotifications.ts [hunk 3]
```diff
@@ -42,44 +44,6 @@ export function useSwitchFeatureAnnouncementsChange(): {
 
 export type UseSwitchAccountNotificationsData = { [address: string]: boolean };
 
-export function useSwitchAccountNotifications(): {
```
Comment (Prithpal-Sooriya) [inline ui/hooks/metamask-notifications/useSwitchNotifications.ts:45]: This was never used, so removed.

File: ui/hooks/metamask-notifications/useSwitchNotifications.ts [hunk 4]
```diff
@@ -169,15 +133,12 @@ export function useAccountSettingsProps(accounts: string[]) {
...
   // Effect - async get if accounts are enabled/disabled
   useEffect(() => {
-    try {
-      const memoAccounts: string[] = JSON.parse(jsonAccounts);
-      update(memoAccounts);
-    } catch {
-      setError('Failed to get account settings');
-    } finally {
-      setLoading(false);
+    if (!isEnabled) {
```
Comment (Prithpal-Sooriya) [inline ui/hooks/metamask-notifications/useSwitchNotifications.ts:136]: Add enable check to bail early if not enabled.

File: ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx [hunk 5]
```diff
@@ -74,7 +74,7 @@ export function NotificationsSettingsAllowNotifications({
   }, [isMetamaskNotificationsEnabled]);
 
   useEffect(() => {
-    if (!error) {
+    if (!error && isMetamaskNotificationsEnabled) {
```
Comment (Prithpal-Sooriya) [inline ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx:77]: Add additional check to only run content if enabled.

Comment (Prithpal-Sooriya) [reply ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx:77]: I mentioned in the description that I dislike the multiple ~useFootgun~ useEffect's spread around in a bunch of components. I'll do a larger refactor and cleanup.

**Review by cmd-ob — APPROVED** [review]

**Pull request merged**